### PR TITLE
Active job to perform background jobs

### PR DESCRIPTION
**Pull request by toothfairy**
Replaces Sidekiq queue backend with ActiveJob interface

### CODE REVIEW

**Comment by sindrenm** [comment]
Is there a reason you're renaming the AsyncMessanger to AsyncJob?

**Comment by toothfairy** [comment]
@sindrenm yes kind of, it doesn't really manage anything, it's rather actor which performs something. I though it's more correct semantically

**Comment by sindrenm** [comment]
And what about the `deliver_later` method being renamed to `deliver_async`? I specifically called it `deliver_later` to make it feel similar to `perform_later` in ActiveJob.

**Comment by sindrenm** [comment]
For reference, see this comment:
https://github.com/hyperoslo/cellular/pull/12#discussion_r18326525

File: lib/cellular/models/sms.rb [hunk 1]
```diff
@@ -21,12 +23,13 @@ def deliver
...
 
-    def deliver_later
-      Cellular::Jobs::AsyncMessenger.perform_async options
+    def deliver_async(delivery_options = {})
+      Cellular::Jobs::AsyncJob.set(delivery_options).perform_later(options)
     end
 
     def deliver_at(timestamp)
-      Cellular::Jobs::AsyncMessenger.perform_at timestamp, options
+      warn "[DEPRECATION] 'deliver_at' is deprecated; use 'deliver_async' instead"
```
Comment (sindrenm) [inline lib/cellular/models/sms.rb]: Since this change is probably backwards incompatible, can't we just remove this method altogether while we're at it?

Comment (toothfairy) [inline lib/cellular/models/sms.rb]: @sindrenm you just asked it in RL :D

Comment (sindrenm) [inline lib/cellular/models/sms.rb]: I know, but now it's documented! Isn't the Internet great? :-D

**Comment by sindrenm** [comment]
All in all, I think this is great. Progress in all things. :grinning:

File: spec/cellular/jobs/async_messenger_spec.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 require 'spec_helper'
 
-describe Cellular::Jobs::AsyncMessenger do
+describe Cellular::Jobs::AsyncJob do
```
Comment (sindrenm) [inline spec/cellular/jobs/async_messenger_spec.rb]: Are you gonna change the filename for this spec as well? :wink:

Comment (toothfairy) [inline spec/cellular/jobs/async_messenger_spec.rb]: yes, I'll revert it back to messenger.

Comment (sindrenm) [inline spec/cellular/jobs/async_messenger_spec.rb]: #winning

**Comment by toothfairy** [comment]
@sindrenm, can you check laters fixes plx

File: lib/cellular/models/sms.rb [hunk 3]
```diff
@@ -41,16 +40,6 @@ def delivered?
...
-    def country
-      warn "[DEPRECATION] 'country' is deprecated; use 'country_code' instead"
-      @country_code
-    end
-
-    def country=(country)
-      warn "[DEPRECATION] 'country' is deprecated; use 'country_code' instead"
-      @country_code = country
-    end
-
```
Comment (sindrenm) [inline lib/cellular/models/sms.rb]: So I'm not entirely sure if this belongs in this PR. But now that it's here, I'm sure we can get away with it. :wink:

Comment (sindrenm) [inline lib/cellular/models/sms.rb]: At least remove the specs that checks for the deprecation warnings and stuff. :smiley:

**Comment by sindrenm** [comment]
Looks good to me. :smiley:

**Pull request merged**